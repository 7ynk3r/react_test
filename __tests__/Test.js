import { View, Button, Text } from 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

// const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

const data = {
  user: 'joe',
  items: [{
    id: 1,
    name: 'item 1',
    price: 1,
  },{
    id: 2,
    name: 'item 2',
    price: 2,
  }],
}

const Cart = ({ cart }) => (
  <View>
    <User user={ cart.user } />
    <Items items={ cart.items } />
  </View>
);

const User = ({ user }) => <Text>{ user }</Text>;

class Items extends React.Component {
  constructor(props) {
    super(props);
    this.state = this.calculateState(props.items);
  }
  calculateState = (items) => ({
    items,
    total: items.map(item => item.value)
      .reduce((a, b) => a + b, 0),
  })
  deleteItem = ({ id }) => {
    const items = this.state.items.filter(item => item.id !== id);
    this.setState(this.calculateState(items));
  }
  render = () => (
    <View>
      { this.state.items.map(item => (
        <Item
          key={ item.id }
          item={ item }
          onDelete={ () => this.deleteItem(item) }
        />
      ))}
      <Total total={ this.state.total }/>
    </View>
  )
}

const Total = ({ total }) => <Text>Total { total }</Text>

const Item = ({ item, onDelete }) => (
  <View>
    <Text>{ item.name }</Text>
    <Text>{ item.price }</Text>
    <Button title='Delete' onPress={ onDelete }/>
  </View>
);

it('renders correctly', () => {
  const cart = {
    user: 'joe',
    items: [{
      id: 1, name: 'item 1', value: 7,
    }, {
      id: 2, name: 'item 2', value: 4,
    }]
  };
  const tree = renderer.create(<Cart cart={ cart }/>);
  expect(tree).toMatchSnapshot();
});


it('renders empty correctly', () => {
  const renderer = new ShallowRenderer();
  const items = [];

  renderer.render(<Items items={ items }/>);
  const result0 = renderer.getRenderOutput();
  expect(result0).toMatchSnapshot();
});

it('renders one item correctly', () => {
  const renderer = new ShallowRenderer();
  const items = [{
    id: 1, name: 'item 1', value: 7,
  }];

  renderer.render(<Items items={ items }/>);
  const result0 = renderer.getRenderOutput();
  expect(result0).toMatchSnapshot();
});

it('renders one item deleted correctly', () => {
  const renderer = new ShallowRenderer();
  const items = [{
    id: 1, name: 'item 1', value: 7,
  }, {
    id: 2, name: 'item 2', value: 4,
  }];

  renderer.render(<Items items={ items }/>);
  const result0 = renderer.getRenderOutput();
  expect(result0).toMatchSnapshot();

  // delete item 2 item.
  result0.props.children[0][0].props.onDelete();
  const result1 = renderer.getRenderOutput();
  expect(result1).toMatchSnapshot();
});
