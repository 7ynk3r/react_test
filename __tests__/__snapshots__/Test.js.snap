// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    joe
  </Text>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        item 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      >
        item 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Total 
      11
    </Text>
  </View>
</View>
`;

exports[`renders empty correctly 1`] = `
<View>
  <Total
    total={0}
  />
</View>
`;

exports[`renders one item correctly 1`] = `
<View>
  <Item
    item={
      Object {
        "id": 1,
        "name": "item 1",
        "value": 7,
      }
    }
    onDelete={[Function]}
  />
  <Total
    total={7}
  />
</View>
`;

exports[`renders one item deleted correctly 1`] = `
<View>
  <Item
    item={
      Object {
        "id": 1,
        "name": "item 1",
        "value": 7,
      }
    }
    onDelete={[Function]}
  />
  <Item
    item={
      Object {
        "id": 2,
        "name": "item 2",
        "value": 4,
      }
    }
    onDelete={[Function]}
  />
  <Total
    total={11}
  />
</View>
`;

exports[`renders one item deleted correctly 2`] = `
<View>
  <Item
    item={
      Object {
        "id": 2,
        "name": "item 2",
        "value": 4,
      }
    }
    onDelete={[Function]}
  />
  <Total
    total={4}
  />
</View>
`;
